### ISSUE TITLE
[amp-next-page] <link> for custom font deleted in following pages

### ISSUE THREAD

**Issue by flyinglimao** [ampproject/amphtml#29283]
## What's the issue?

We recently add amp-next-page for infinite scrolling and found that custom font (here is font-awesome) included by `<link>` didn't shows in second and following pages.

I think this is result from the `<link>` element isn't attached to the shadow DOM generated by amp-next-page.

Original HTML and `<link>` tag:
<img width="946" alt="圖片" src="https://user-images.githubusercontent.com/11978467/87521733-596f4280-c6b7-11ea-80ba-8b0d414af119.png">

AMP-next-page Shadow DOM missing the tag:
<img width="946" alt="圖片" src="https://user-images.githubusercontent.com/11978467/87522052-bc60d980-c6b7-11ea-90fd-c5d9d6175785.png">

If I add `<link>` back, it works (compares to image below):
<img width="946" alt="圖片" src="https://user-images.githubusercontent.com/11978467/87522539-545ec300-c6b8-11ea-8472-99a852051ebc.png">

## How do we reproduce the issue?

Test case: https://womany.net/articles/22068/amp 

1. Open the given page, you can see icons near the date in article title block shows.<img width="1025" alt="圖片" src="https://user-images.githubusercontent.com/11978467/87522858-b91a1d80-c6b8-11ea-96c6-55d829807303.png">

2. Scroll to bottom and let it load next article.
3. Look into same position, the icon disappears.<img width="1074" alt="圖片" src="https://user-images.githubusercontent.com/11978467/87522954-dc44cd00-c6b8-11ea-86bd-8936ae7fb4ed.png">

## What browsers are affected?

All browsers

## Which AMP version is affected?

2007040248002

**Comment by Ichigo3d** [comment]
cc @caroqliu

**Comment by wassgha** [comment]
@dvoytenko might have more guidance here.

Investigated this issue and it seems like we intentionally ignore stylesheets (since AMP does not support them anyway) and assume that having the stylesheet on the parent document is sufficient for the font to be loaded. This is true in most cases (where the stylesheet only contains font definitions), however for fontawesome's case, the stylesheet also contains class definitions (e.g. `.fa-heart`) which do not propagate to the shadow dom (given that it's css-isolated). Not sure how to proceed here, since fontawesome seems to be a special case.

**Comment by dvoytenko** [comment]
IMHO it's ok to put stylesheets on a shadow doc level. But, can we place `<link>` inside a shadow root? I actually think the current implementation was due to some restrictions or maybe bugs.

**Comment by irvin** [comment]
Do we have the stats of how many percents of amp page contains font awesome to know the impact of the issue?

There are [only 5 exceptions](https://amp.dev/documentation/guides-and-tutorials/develop/style_and_layout/custom_fonts/#1.-using-%3Clink%3E) on <link> to stylesheets, perhaps we should also add them as an exception and make them available in shadow dom to easily fix the issue.

**Comment by wassgha** [comment]
[This](https://www.smashingmagazine.com/2016/12/styling-web-components-using-a-shared-style-sheet/) is a good summary of the problem. It seems like `<link/>` in ShadowDOM was added back to the spec and currently works in Chrome (need to test on other browsers). Alternatively, `<style>@import url</style>` is a viable alternative but seems to have some performance issues associates. We can obviously hardcode this but that wouldn't be future-proof.

**Comment by ydnikolaev** [comment]
same here
since I use font-awesome in many parts of site — it's really hurts

**Comment by dvoytenko** [comment]
IMHO `link` or `@import` sound ok to me.
